tests/unit/aiplatform/test_featurestores.py

```diff
@@ -1054,6 +1218,200 @@ def test_ingest_from_gcs_with_invalid_gcs_source_type(self):
+        )
+        assert type(result) == pd.DataFrame
+        assert len(result) == 1
+        assert result.entity_id[0] == _TEST_READ_ENTITY_ID
+        assert result.get(_TEST_FEATURE_ID)[0] == _TEST_FEATURE_VALUE
+
+    @pytest.mark.parametrize(
+        "feature_ids, feature_value_types, entity_ids, feature_values, expected_df",
+        [
+            (
+                _TEST_FEATURE_IDS_FOR_DF_CONSTRUCTION,
+                _TEST_FEATURE_VALUE_TYPES_FOR_DF_CONSTRUCTION,
+                ["entity_01", "entity_02"],
+                [
+                    [
+                        False,
+                        [True, False],
+                        1.2,
+                        [1.2, 3.4],
+                        1,
+                        [1, 2],
+                        "test",
+                        ["test1", "test2"],
+                        b"1",
+                    ],
+                    [
+                        True,
+                        [True, True],
+                        2.2,
+                        [2.2, 4.4],
+                        2,
+                        [2, 3],
+                        "test1",
+                        ["test2", "test3"],
+                        b"0",
+                    ],
+                ],
+                pd.DataFrame(
+                    data=[
+                        [
+                            "entity_01",
+                            False,
+                            [True, False],
+                            1.2,
+                            [1.2, 3.4],
+                            1,
+                            [1, 2],
+                            "test",
+                            ["test1", "test2"],
+                            b"1",
+                        ],
+                        [
+                            "entity_02",
+                            True,
+                            [True, True],
+                            2.2,
+                            [2.2, 4.4],
+                            2,
+                            [2, 3],
+                            "test1",
+                            ["test2", "test3"],
+                            b"0",
+                        ],
+                    ],
+                    columns=["entity_id"] + _TEST_FEATURE_IDS_FOR_DF_CONSTRUCTION,
+                ),
+            ),
+            (
+                _TEST_FEATURE_IDS_FOR_DF_CONSTRUCTION,
+                _TEST_FEATURE_VALUE_TYPES_FOR_DF_CONSTRUCTION,
+                ["entity_01", "entity_02"],
+                [
+                    [
+                        False,
+                        [True, False],
+                        1.2,
+                        [1.2, 3.4],
+                        1,
+                        [1, 2],
+                        "test",
+                        ["test1", "test2"],
+                        b"1",
+                    ],
+                    [None, None, None, None, None, None, None, None, None],
```

Comment:
nit. Can you change/add a row that have both valid values and None.

Reply:
updated